crates/polars-core/src/series/from.rs

```diff
@@ -517,37 +517,33 @@ unsafe fn to_physical_and_dtype(
             to_physical_and_dtype(out, md)
         },
         #[cfg(feature = "dtype-array")]
-        #[allow(unused_variables)]
         ArrowDataType::FixedSizeList(field, size) => {
-            feature_gated!("dtype-array", {
```

Comment:
drive-by: This is already feature-gated, no real sense in feature gating it again